internal/internal_task_pollers.go

```diff
@@ -765,12 +765,27 @@ func (wtp *workflowTaskPoller) poll(ctx context.Context) (interface{}, error) {
 
 	response, err := wtp.service.PollForDecisionTask(ctx, request, getYarpcCallOptions(wtp.featureFlags)...)
```

Comment:
Do you think we should use `backoff.Retry` like other endpoints? In this way, we don't have to expose `ErrRetryableAfter` to the outside and it's more consistent across all endpoints

Reply:
there's an exponential backoff at a higher level, this is within that retry.

it may be easier to change that retryer to check the error value though 🤔.  I think this one was ignoring a number of categories, but I'm not 100% sure why.

Reply:
right. Let's land it first. This refactoring can come later

Reply:
Update!

Nope.  The poller uses a unique kind of ratelimiter and retry loop :|  Changing that is substantial and unifying the two strategies is probably not going to be an improvement, at least without a bigger re-thinking of retries and ratelimiting in general.